js/gravityforms-mfi-multifile.js

```diff
@@ -0,0 +1,290 @@
+
+                $('#' + up.settings.filelist).prepend(status);
+                totalCount++;
+
+            });
+
+            up.refresh(); // Reposition Flash
+
+            var formElementID = "form#gform_" + formID;
+            var uidElementID = "input:hidden[name='gform_unique_id']";
+            var uidSelector = formElementID + " " + uidElementID;
+            var $uid = $(uidSelector);
+            if($uid.length==0){
+                $uid = $(uidElementID);
+            }
+
+            uniqueID = $uid.val();
+            if('' === uniqueID){
+                uniqueID = generateUniqueID();
+                $uid.val(uniqueID);
+            }
+
+
+            if(max > 0 && totalCount >= max){
+                gfMultiFileUploader.toggleDisabled(up.settings, true);
+                addMessage(up.settings.gf_vars.message_id, strings.max_reached)
+            }
+
+
+            up.settings.multipart_params.gform_unique_id = uniqueID;
+            up.start();
+
+        });
+
+        uploader.bind('UploadProgress', function(up, file) {
+            var html = file.percent + "%";
+            $('#' + file.id + " b").html(html);
+        });
+
+        uploader.bind('Error', function(up, err) {
+            if(err.code === plupload.FILE_EXTENSION_ERROR){
+                var extensions = typeof up.settings.filters.mime_types != 'undefined' ? up.settings.filters.mime_types[0].extensions /* plupoad 2 */ : up.settings.filters[0].extensions;
+                addMessage(up.settings.gf_vars.message_id, err.file.name + " - " + strings.invalid_file_extension + " " + extensions);
+            } else if (err.code === plupload.FILE_SIZE_ERROR) {
+                addMessage(up.settings.gf_vars.message_id, err.file.name + " - " + strings.file_exceeds_limit);
+            } else {
+                var m = "<li>Error: " + err.code +
+                    ", Message: " + err.message +
+                    (err.file ? ", File: " + err.file.name : "") +
+                    "</li>";
+
+                addMessage(up.settings.gf_vars.message_id, m);
+            }
+            $('#' + err.file.id ).html('');
+
+            up.refresh(); // Reposition Flash
+        });
+
+        uploader.bind('FileUploaded', function(up, file, result) {
+            var response = $.secureEvalJSON(result.response);
+            if(response.status == "error"){
+                addMessage(up.settings.gf_vars.message_id, file.name + " - " + response.error.message);
+                $('#' + file.id ).html('');
+                return;
+            }
+
+            var html = '<strong>' + file.name + '</strong>';
+            var formId = up.settings.multipart_params.random_id;
+            var fieldId = up.settings.multipart_params.field_id;
+            html = "<img "
+                + "class='gform_delete' "
+                + "src='" + imagesUrl + "/delete.png' "
+                + "onclick='gformDeleteUploadedFile(" + formId + "," + fieldId + ", this);' "
```

Comment:
Here and on the next line, the formID is, as everywhere else, the random form id assigned by this plugin.
This allows proper deletion of the uploaded file.